app/api/routes/image_queries.py

```diff
@@ -202,6 +212,10 @@ async def post_image_query(
             }
         )
 
+        # Fail if edge inference is not available and edge-only mode is enabled
+        if edge_only:
+            raise RuntimeError("EDGE_ONLY is set to ENABLED, but edge inference is not available.")
```

Comment:
I'm not 100% confident that this is the right place in the logic for this, so lmk if this seems wrong.

Reply:
Does this just stops the program or you are handling the runtime error somewhere?

Reply:
Right now it just stops the program. I wasn't sure what the best thing to do here is because there's the use case where the first time you submit an IQ to a detector, it will create the edge inference pod for it. That's why I put this below the `create_inference_deployment_record` call, so that the new pod will still get triggered even once the program stops. If you think there's a better way to handle this though I'm definitely down to change the behavior.

Reply:
I think from an API perspective this should look like a 500 error. IDK if fastapi is smart enough to do the translation to that on its own or if we need to explicitly handle it.

Reply:
Just tested it and it does give a 500 response, with `Reason: Internal Server Error`.

Reply:
Yea for me I think it should automatically launch the pod and just revert to cloud API calls. Maybe use logger.warning to say that it is still waiting for inference pod to launch so it has to use cloud calls? 

But this way will still be slowing down the app if the inference pod is not ready. So I am not 100% sure if giving out 500 response to the client vs. just give cloud answer with an delay is a good solution.

Reply:
I dont think we should revert to calling the cloud, thats doing explicitly what the user said NOT to do.

Reply:
They could have good reason for not wanting to go to the cloud, such as cost